jamaljsr/polar
# feat(litd): Add full support for decimal display for TAP assets

Closes #986 
Depends on #1039 

### Description

Adds support for setting the `decimal_display` field when minting assets. I've also updated all of the labels and input fields to support decimal values when creating addresses, sending assets on-chain, funding asset channels, creating asset invoices, and paying asset invoices.

### Steps to Test

First, you'll need to build the docker image using the litd PR branch:
```
$ cd docker/litd/
$ docker build --build-arg LITD_VERSION=update-to-lnd-18-4 -t polarlightning/litd:0.14.0-pre1 -f src.Dockerfile --no-cache .
```

Then in Polar:
1. Create a new network with 4 `litd` nodes
2. Mint a PUSD asset with a 100,050 supply using 2 decimal places
3. Create an 500.25 PUSD asset address on `carol`
4. Send PUSD to that address from `alice`
5. Open a 400.50 PUSD asset channel from `alice` to `bob`
6. Open a 100M sats channel from `bob` to `carol`
7. Open a 400.50 PUSD asset channel from `carol` to `dave`
8. Create a 200.25 PUSD asset invoice on `dave`
9. Pay the invoice on `alice` using PUSD
10. Confirm the payment completes successfully
11. Confirm the asset amount labels throughout the app all show values with 2 decimal places

### Screenshots

![image](https://github.com/user-attachments/assets/de77477e-326e-40ab-b29e-ae32cfc50f40)
